### ISSUE TITLE
Auto indent outside blocks for hcl

### ISSUE THREAD

**Issue by wcarlsen** [nvim-treesitter/nvim-treesitter#2590]
### Describe the bug

When adding new line auto indent works great when inside a block, but when outside of a block it also auto indent when new line is added.

**Screenshots**
Hard to show on screenshots. But here is an attempt. See it on line 5.
![image](https://user-images.githubusercontent.com/17003032/155498337-b0ff556a-2f49-486c-9d57-17e0bc7fcb8c.png)

### To Reproduce

1. I use stock [AstroVim](https://github.com/kabinspace/AstroVim) with :TSInstall hcl
2. Write any Terraform block and try inserting a new line after that block. And see it auto indent.

### Expected behavior

I expect auto indent not to happen out side Terraform blocks.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.4 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v17.6.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (GCC) 11.2.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - hcl            ✓ . ✓ ✓ ✓ 
  - bash           ✓ ✓ ✓ . ✓ 
  - yaml           ✓ ✓ ✓ ✓ ✓ 
  - lua            ✓ ✓ ✓ ✓ ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.6.1
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

### Additional context

 Also disabling indent makes it work outside Terraform blocks, but breaks indentation inside blocks. This issue is moved up to this project by suggestion of the maintainer of this project https://github.com/MichaHoffmann/tree-sitter-hcl/issues/15.

**Comment by kyazdani42** [comment]
I've reproduced. The algorithm for indentation has been improved a bit lately, there are some things to add i think now to make this work. I might fix this if i can make the tests work on my machine :smile:

**Comment by MichaHoffmann** [comment]
Hey @kyazdani42 thanks for looking into it, if you need some help i'd be glad to.

**Comment by MichaHoffmann** [comment]
probably can be closed by #2855
